src/components/views/rooms/RoomTile.js

```diff
@@ -126,36 +135,39 @@ module.exports = React.createClass({
         this.setState( { badgeHover : false } );
     },
 
-    onBadgeClicked: function(e) {
-        // Only allow none guests to access the context menu
-        if (!MatrixClientPeg.get().isGuest()) {
+    showContextMenu: function() {
+        // Hide the tooltip
+        if (this.props.collapsed) {
+            this.setState({ hover: false });
+        }
 
-            // If the badge is clicked, then no longer show tooltip
-            if (this.props.collapsed) {
-                this.setState({ hover: false });
+        const RoomTileContextMenu = sdk.getComponent('context_menus.RoomTileContextMenu');
+        const elementRect = this.refs.badge.getBoundingClientRect();
+
+        // The window X and Y offsets are to adjust position when zoomed in to page
+        const x = elementRect.right + window.pageXOffset + 3;
+        const chevronOffset = 12;
+        let y = (elementRect.top + (elementRect.height / 2) + window.pageYOffset);
+        y = y - (chevronOffset + 8); // where 8 is half the height of the chevron
+
+        ContextualMenu.createMenu(RoomTileContextMenu, {
+            chevronOffset: chevronOffset,
+            left: x,
+            top: y,
+            closeOnMouseLeave: true,
```

Comment:
Is this meant to be false if the menu was opened with a click or something? It seems to be an option that's only ever set to true.